[18.03] firefox-{esr,} 60

andir opened the pull request:
###### Motivation for this change

Backport of #40006 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
sqlite: 3.21.0 -> 3.22.0

(cherry picked from commit 6450c1941991c219e4388a69328e050c65cd861f)

Commit:
sqlite: Raising max variable number and max expr depth values

fixes #36293

(cherry picked from commit fe5f012d074e9d7fb332288b53bd90b8587bf134)

oxij committed:
firefoxPackages: add a patch to fix pulseaudio initialization bug

This is optional (`libcardiacarrest` has a workaround for this bug
because there's `firefox-bin` that I can't fix), but with this applied things
are a bit smoother.

taku0 committed:
firefox-bin: 59.0.2 -> 59.0.3

(cherry picked from commit 8a16d53897cf6d6e1a8f01a9961438e821099d71)

taku0 committed:
firefox: 59.0.2 -> 59.0.3

(cherry picked from commit 632f437e5604ef33796e12dfd7fdf6c61b622f93)

taku0 committed:
firefox-esr: 52.7.3esr -> 52.7.4esr

(cherry picked from commit c259ecc04e77a0783b6be49c5b97bd44ab555154)

taku0 committed:
firefox: use fetchzip instead of fetchurl

(cherry picked from commit 1a708fc0633a8d5e2477e7e5c8d3d70baf951ab2)

taku0 committed:
firefox-bin: 59.0.3 -> 60.0

(cherry picked from commit 2872345b9b3e00939fea4c7ddf1b7a3d2b9ee9db)

taku0 committed:
nss: 3.35 -> 3.36.1

(cherry picked from commit e96473b24982694efe9c1f2f10e01310a01750b7)

taku0 committed:
nspr: 4.18 -> 4.19

(cherry picked from commit 05a2f8f4a7be373153f95e07ee4625d77e41095b)

taku0 committed:
firefox: 59.0.3 -> 60.0

(cherry picked from commit 5a15da04b8101324b15a02ce1a6216ecd9a4e1dc)

taku0 committed:
firefox-esr: 52.7.4esr -> 52.8.0esr, 60.0esr

(cherry picked from commit b2725027d63eb6a507a4adfbdcbabc453d7071bd)

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=83083873-df14-4766-a5d2-96dbdb7292c2)

Attempted: nspr, nss

<details><summary>Partial log (click to expand)</summary><p>

```
patching file nss/lib/ckfw/pem/psession.c
patching file nss/lib/ckfw/pem/pslot.c
patching file nss/lib/ckfw/pem/ptoken.c
patching file nss/lib/ckfw/pem/rsawrapr.c
patching file nss/lib/ckfw/pem/util.c
patching file lib/freebl/Makefile
Hunk #1 FAILED at 522.
1 out of 1 hunk FAILED -- saving rejects to file lib/freebl/Makefile.rej
builder for '/nix/store/mzvc9r1dwgh07r1r61288q1q40bwvcb1-nss-3.36.1.drv' failed with exit code 1
[31;1merror:[0m build of '/nix/store/mzvc9r1dwgh07r1r61288q1q40bwvcb1-nss-3.36.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=408ab06f-0b2f-465a-86f8-7e1e2729a1c1)

Attempted: nspr, nss

<details><summary>Partial log (click to expand)</summary><p>

```
    bb e1 3f eb d5 99 66 ab fa db
    4b d5 e9 70
moduleSpec configdir='' certPrefix='' keyPrefix='' secmod='' flags=noCertDB, noModDB
Generate a DSA key pair ...
Moving /nix/store/3mbff5r06yj5kbgysr7j7fh17bwgyjxq-nss-3.36.1/bin to /nix/store/m6in1g7hvyqnzqy01c17326xmgbqciq5-nss-3.36.1-tools/bin
rmdir: failed to remove '/nix/store/3mbff5r06yj5kbgysr7j7fh17bwgyjxq-nss-3.36.1': Directory not empty
Moving /nix/store/m6in1g7hvyqnzqy01c17326xmgbqciq5-nss-3.36.1-tools/bin/nss-config to /nix/store/j3yjlkq894fby4wwf0zpjnicqx87isny-nss-3.36.1-dev/bin/nss-config
rmdir: failed to remove '/nix/store/m6in1g7hvyqnzqy01c17326xmgbqciq5-nss-3.36.1-tools/bin': Directory not empty
Moving /nix/store/3mbff5r06yj5kbgysr7j7fh17bwgyjxq-nss-3.36.1/lib/libcrmf.a to /nix/store/j3yjlkq894fby4wwf0zpjnicqx87isny-nss-3.36.1-dev/lib/libcrmf.a
rmdir: failed to remove '/nix/store/3mbff5r06yj5kbgysr7j7fh17bwgyjxq-nss-3.36.1/lib': Directory not empty
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=8cf34a14-56b5-4d54-a02e-455512d2b3d5)

Attempted: nspr, nss

<details><summary>Partial log (click to expand)</summary><p>

```
moduleSpec configdir='' certPrefix='' keyPrefix='' secmod='' flags=noCertDB, noModDB
Generate a DSA key pair ...
Moving /nix/store/4hw5zb26hwqva0f2rghbkmsan4fi9212-nss-3.36.1/bin to /nix/store/q8qrj23zkkfnhzw0apiracsshk1w7850-nss-3.36.1-tools/bin
rmdir: failed to remove '/nix/store/4hw5zb26hwqva0f2rghbkmsan4fi9212-nss-3.36.1': Directory not empty
Moving /nix/store/q8qrj23zkkfnhzw0apiracsshk1w7850-nss-3.36.1-tools/bin/nss-config to /nix/store/bng930ck241l44fabkdp7xkzyna66pjn-nss-3.36.1-dev/bin/nss-config
rmdir: failed to remove '/nix/store/q8qrj23zkkfnhzw0apiracsshk1w7850-nss-3.36.1-tools/bin': Directory not empty
Moving /nix/store/4hw5zb26hwqva0f2rghbkmsan4fi9212-nss-3.36.1/lib/libcrmf.a to /nix/store/bng930ck241l44fabkdp7xkzyna66pjn-nss-3.36.1-dev/lib/libcrmf.a
rmdir: failed to remove '/nix/store/4hw5zb26hwqva0f2rghbkmsan4fi9212-nss-3.36.1/lib': Directory not empty
/nix/store/833hms4620zqlhw5ggxza69kp98q8sal-nspr-4.19
/nix/store/4hw5zb26hwqva0f2rghbkmsan4fi9212-nss-3.36.1
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=baae2efa-3f57-47a2-92eb-175a2ee1fffd)

Attempted: nspr, nss

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ryhr1ljccqcnj7qrrwdb6xdqcmg0391s-nspr-4.19
/nix/store/3mbff5r06yj5kbgysr7j7fh17bwgyjxq-nss-3.36.1
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=eb8c2dd2-88a3-4ca0-a4ba-62ccde042d67)

Attempted: nspr, nss

<details><summary>Partial log (click to expand)</summary><p>

```
patching file nss/lib/ckfw/pem/psession.c
patching file nss/lib/ckfw/pem/pslot.c
patching file nss/lib/ckfw/pem/ptoken.c
patching file nss/lib/ckfw/pem/rsawrapr.c
patching file nss/lib/ckfw/pem/util.c
patching file lib/freebl/Makefile
Hunk #1 FAILED at 522.
1 out of 1 hunk FAILED -- saving rejects to file lib/freebl/Makefile.rej
builder for '/nix/store/mzvc9r1dwgh07r1r61288q1q40bwvcb1-nss-3.36.1.drv' failed with exit code 1
[31;1merror:[0m build of '/nix/store/mzvc9r1dwgh07r1r61288q1q40bwvcb1-nss-3.36.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=65f7320f-6ef4-4fce-af40-e1555ba0b212)

Attempted: nspr, nss

<details><summary>Partial log (click to expand)</summary><p>

```
moduleSpec configdir='' certPrefix='' keyPrefix='' secmod='' flags=noCertDB, noModDB
Generate a DSA key pair ...
Moving /nix/store/0v2wsbdjcrxvj9iqvahzsfyj9xszbkcv-nss-3.36.1/bin to /nix/store/w6rdpi2znci6z0wjb4wq61gmp686zg3h-nss-3.36.1-tools/bin
rmdir: failed to remove '/nix/store/0v2wsbdjcrxvj9iqvahzsfyj9xszbkcv-nss-3.36.1': Directory not empty
Moving /nix/store/w6rdpi2znci6z0wjb4wq61gmp686zg3h-nss-3.36.1-tools/bin/nss-config to /nix/store/k1imnpsw08wspyvh2i0mjnxn4xxnihjy-nss-3.36.1-dev/bin/nss-config
rmdir: failed to remove '/nix/store/w6rdpi2znci6z0wjb4wq61gmp686zg3h-nss-3.36.1-tools/bin': Directory not empty
Moving /nix/store/0v2wsbdjcrxvj9iqvahzsfyj9xszbkcv-nss-3.36.1/lib/libcrmf.a to /nix/store/k1imnpsw08wspyvh2i0mjnxn4xxnihjy-nss-3.36.1-dev/lib/libcrmf.a
rmdir: failed to remove '/nix/store/0v2wsbdjcrxvj9iqvahzsfyj9xszbkcv-nss-3.36.1/lib': Directory not empty
/nix/store/zmqj9x444chy98avj5sf7i4vn3mgy51b-nspr-4.19
/nix/store/0v2wsbdjcrxvj9iqvahzsfyj9xszbkcv-nss-3.36.1
```
</p></details>

andir committed:
nss: removed aarch64 patch

With the update to 3.36 the patch introduced with e96473b is no longer
required.

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=17ad7e9d-61ea-424c-bf99-38a73fa4937e)

Attempted: nspr, nss, sqlite

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/8nn6whdfxk5jslnv1z6ydvwri8hqh6zd-apple-framework-NetFS.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/nw53nah3jvbihg844zwk774bjiriqngb-apple-framework-OpenDirectory.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/ika32aq4vwjkrdnybcdjfllh6pdg9l34-apple-framework-CoreAudio.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/6khy3mjc1b3dai8ind11yw1w92zxhh5l-apple-framework-DiskArbitration.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/snq9331qq3p2xx99xs0ayjbs2h27dcpq-apple-framework-Security.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/64n1qs06xck6x98jcq7zim9mb35995dh-apple-framework-ServiceManagement.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/0bf6x63kxx7c8pr4jrx7acnc9aq166mb-apple-framework-CoreServices.drv': 12 dependencies couldn't be built
cannot build derivation '/nix/store/b39zyn18gcdbcqgmhnyqfmcnclv4a2m8-nspr-4.19.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/inm6zp52382b0h32xpqq4lancx6lmkk2-nss-3.36.1.drv': 7 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/7yv64rs195gn93kwk39drw2l0fsqcaf8-sqlite-3.22.0.drv', '/nix/store/b39zyn18gcdbcqgmhnyqfmcnclv4a2m8-nspr-4.19.drv', '/nix/store/inm6zp52382b0h32xpqq4lancx6lmkk2-nss-3.36.1.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=c7a0f739-fe7b-4003-ade8-4cb375d82195)

Attempted: nspr, nss, sqlite

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ryhr1ljccqcnj7qrrwdb6xdqcmg0391s-nspr-4.19
/nix/store/vgplmqd3wymass20b019lvkgavis5xya-nss-3.36.1
/nix/store/0dzaiissi1ch89g74c29fr53m4kiryim-sqlite-3.22.0-bin
```
</p></details>

GrahamcOfBorg commented:
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=9fc6efb4-e58e-4e44-b9a0-1ba60a913419)

Attempted: nspr, nss, sqlite

<details><summary>Partial log (click to expand)</summary><p>

```
Generate a DSA key pair ...
Moving /nix/store/nx0s7s7pddny8x10j628a18p8jw9ccsa-nss-3.36.1/bin to /nix/store/swrax095kcf3f9kpj40x6v76yc2c6kll-nss-3.36.1-tools/bin
rmdir: failed to remove '/nix/store/nx0s7s7pddny8x10j628a18p8jw9ccsa-nss-3.36.1': Directory not empty
Moving /nix/store/swrax095kcf3f9kpj40x6v76yc2c6kll-nss-3.36.1-tools/bin/nss-config to /nix/store/j3xdj5kpw1v9fhds12imlypb8nw86cjz-nss-3.36.1-dev/bin/nss-config
rmdir: failed to remove '/nix/store/swrax095kcf3f9kpj40x6v76yc2c6kll-nss-3.36.1-tools/bin': Directory not empty
Moving /nix/store/nx0s7s7pddny8x10j628a18p8jw9ccsa-nss-3.36.1/lib/libcrmf.a to /nix/store/j3xdj5kpw1v9fhds12imlypb8nw86cjz-nss-3.36.1-dev/lib/libcrmf.a
rmdir: failed to remove '/nix/store/nx0s7s7pddny8x10j628a18p8jw9ccsa-nss-3.36.1/lib': Directory not empty
/nix/store/lxv3zk13mvpsnq6kxq1qdckyj6p3airr-nspr-4.19
/nix/store/nx0s7s7pddny8x10j628a18p8jw9ccsa-nss-3.36.1
/nix/store/7wribi9yrwl5ngj3k9qc75a7phmbcns4-sqlite-3.22.0-bin
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40396&attempt_id=68e20433-d66f-45aa-8f92-b55501b43c6c)

Attempted: nspr, nss, sqlite

<details><summary>Partial log (click to expand)</summary><p>

```
cannot build derivation '/nix/store/8nn6whdfxk5jslnv1z6ydvwri8hqh6zd-apple-framework-NetFS.drv': 4 dependencies couldn't be built
cannot build derivation '/nix/store/nw53nah3jvbihg844zwk774bjiriqngb-apple-framework-OpenDirectory.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/ika32aq4vwjkrdnybcdjfllh6pdg9l34-apple-framework-CoreAudio.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/6khy3mjc1b3dai8ind11yw1w92zxhh5l-apple-framework-DiskArbitration.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/snq9331qq3p2xx99xs0ayjbs2h27dcpq-apple-framework-Security.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/64n1qs06xck6x98jcq7zim9mb35995dh-apple-framework-ServiceManagement.drv': 5 dependencies couldn't be built
cannot build derivation '/nix/store/0bf6x63kxx7c8pr4jrx7acnc9aq166mb-apple-framework-CoreServices.drv': 12 dependencies couldn't be built
cannot build derivation '/nix/store/b39zyn18gcdbcqgmhnyqfmcnclv4a2m8-nspr-4.19.drv': 3 dependencies couldn't be built
cannot build derivation '/nix/store/inm6zp52382b0h32xpqq4lancx6lmkk2-nss-3.36.1.drv': 7 dependencies couldn't be built
[31;1merror:[0m build of '/nix/store/7yv64rs195gn93kwk39drw2l0fsqcaf8-sqlite-3.22.0.drv', '/nix/store/b39zyn18gcdbcqgmhnyqfmcnclv4a2m8-nspr-4.19.drv', '/nix/store/inm6zp52382b0h32xpqq4lancx6lmkk2-nss-3.36.1.drv' failed
```
</p></details>
